### PR TITLE
feat(calimero-sdk): login and auth components for peer apps and admin ui

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(calimero-sdk): login and auth components for peer apps and admin ui
 
## Summary:
Add functions and components for auth and login for both near and metamask.

Fixes # [C20-58](https://calimero.atlassian.net/browse/C20-58)

### CODE REVIEW

File: packages/calimero-sdk/src/wallet/LoginSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+const LoginSelector: React.FC<LoginSelectorProps> = ({
+  applicationId,
+  rpcBaseUrl,
+  successRedirect,
+  cardBackgroundColor,
+  metamaskTitleColor,
+  nearTitleColor,
+  network,
+}) => {
+  const [walletSelected, setWalletSelected] = useState<WalletType | null>(null);
```
Comment (frdomovic) [inline packages/calimero-sdk/src/wallet/LoginSelector.tsx]: @MatejVukosav to make this work like only peers example where wallets were divided on separate pages I can add reading from local storage to get selected wallet state. Or I can remove this part from component and use this "Login selector" just as "redirect to login option page" component. What do you think?

File: examples/only-peers-simple/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "redev": "rm -rf node_modules && pnpm install && vite"
   },
   "dependencies": {
-    "@calimero-is-near/calimero-p2p-sdk": "0.0.11",
+    "@calimero-is-near/calimero-p2p-sdk": "file:../../packages/calimero-sdk",
```
Comment (frdomovic) [inline examples/only-peers-simple/package.json]: will clear later

File: packages/calimero-sdk/src/wallet/MetamaskLogin/Metamask.tsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { randomBytes } from "crypto";
+import { getOrCreateKeypair } from "../../crypto/ed25519";
+import {
+  MetaMaskButton,
+  useAccount,
+  useSDK,
+  // @ts-ignore: sdk-react-ui does not export useSignMessage
+  useSignMessage,
+} from "@metamask/sdk-react-ui";
```
Comment (frdomovic) [inline packages/calimero-sdk/src/wallet/MetamaskLogin/Metamask.tsx:10]: @MatejVukosav not sure how this didn't throw error in only-peers. Using the same metamask version ( 0.18.0 )

File: packages/calimero-sdk/src/wallet/LoginSelector.tsx (hunk 1 above)
Comment (MatejVukosav) [reply packages/calimero-sdk/src/wallet/LoginSelector.tsx]: Second approach is better.  It is just a modal to login with wallet and next step is wallet UI.

File: packages/calimero-sdk/src/wallet/MetamaskLogin/Metamask.tsx (hunk 3 above)
Comment (MatejVukosav) [reply packages/calimero-sdk/src/wallet/MetamaskLogin/Metamask.tsx:10]: idk :)

File: examples/only-peers-simple/src/App.jsx [hunk 4]
```diff
@@ -112,6 +113,19 @@ class App extends React.Component {
...
+        <div style={{
+          display: "flex",
+          width: "100%",
+          justifyContent: "center",
+          alignItems: "center",
+        }}>
+          <LoginSelector
+            applicationId="9SFTEoc6RBHtCn9b6cm4PPmhYzrogaMCd5CRiYAQichP"
+            rpcBaseUrl="http://localhost:2428"
+            successRedirect={() => console.log("success")}
```
Comment (MatejVukosav) [inline examples/only-peers-simple/src/App.jsx]: What about failure?

Comment (frdomovic) [reply examples/only-peers-simple/src/App.jsx]: On failure for it shows you error and you stay on the same page or you can go back to select new wallet account, or wallet

File: packages/calimero-sdk/src/wallet/LoginSelector.tsx (hunk 1 above)
Comment (frdomovic) [reply packages/calimero-sdk/src/wallet/LoginSelector.tsx]: https://github.com/calimero-network/core/assets/93442516/e332479a-854d-4448-85a5-a80d42329505

File: examples/only-peers-simple/src/App.jsx (hunk 4 above)
Comment (MatejVukosav) [reply examples/only-peers-simple/src/App.jsx]: Can you also expose that? Just callback to failure so the app can decide what do to next.

File: packages/calimero-sdk/src/wallet/LoginSelector.tsx (hunk 1 above)
Comment (MatejVukosav) [reply packages/calimero-sdk/src/wallet/LoginSelector.tsx]: Looks good. Why metamask didn't work the first time? In video you clicked multiple times on connect

Comment (frdomovic) [reply packages/calimero-sdk/src/wallet/LoginSelector.tsx]: Chrome bug or whatever, the popup was opened and closed so the button click didn't open it again. Not the FE issue

**Pull request merged**